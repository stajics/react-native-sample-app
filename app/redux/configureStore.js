import { applyMiddleware, createStore, compose } from 'redux';
// middleware
import createLogger from 'redux-logger';
import thunk from 'redux-thunk';
import authenticatedApiCall from './middleware/authenticatedApiCall';
import apiCall from './middleware/apiCall';
import rootReducer from './rootReducer';
// import sessioning from './utils/middleware/sessioning';

const logger = createLogger();
const middleware = [thunk, apiCall, authenticatedApiCall, logger];

const configureStore = (initialState = undefined) => {
  const store = compose(applyMiddleware(...middleware))(createStore)(rootReducer, initialState);
  return store;
};

export default configureStore;
