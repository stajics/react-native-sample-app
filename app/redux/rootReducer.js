import { combineReducers } from 'redux';
// reducers
import { reducer as formReducer } from 'redux-form';
import authentication from '../containers/authentication/reducer';
import home from '../containers/home/reducer';
import profile from '../containers/profile/reducer';
import settings from '../containers/settings/reducer';
import routes from './reducers/routesReducer';
import flags from './reducers/flagsReducer';

export default combineReducers({
  form: formReducer,
  authentication,
  home,
  profile,
  settings,
  routes,
  flags,
});
