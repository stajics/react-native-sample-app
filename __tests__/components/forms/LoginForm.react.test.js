import 'react-native';
import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import LoginForm from '../../../app/components/forms/LoginForm.react'; // eslint-disable-line
const store = createStore(() => ({}));
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'; // eslint-disable-line

it('renders correctly', () => {
  const loginForm = renderer.create(
    <Provider store={store}>
      <LoginForm handleSubmit={jest.fn} />
    </Provider>
  ).toJSON();
  expect(loginForm).toMatchSnapshot();
});
