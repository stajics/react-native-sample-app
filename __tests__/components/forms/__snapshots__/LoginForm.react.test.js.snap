exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": -1,
    }
  }>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
        },
        Object {
          "marginLeft": 10,
          "marginRight": 10,
        },
        undefined,
        Object {
          "flexDirection": "row",
        },
      ]
    }>
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="username"
      style={
        Array [
          Object {
            "backgroundColor": "white",
          },
          Object {
            "height": 40,
            "width": 400,
          },
          Object {},
          Object {
            "flex": -1,
            "textAlign": "left",
            "textAlignVertical": "top",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="" />
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
        },
        Object {
          "marginLeft": 10,
          "marginRight": 10,
        },
        undefined,
        Object {
          "flexDirection": "row",
        },
      ]
    }>
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="password"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
          },
          Object {
            "height": 40,
            "width": 400,
          },
          Object {},
          Object {
            "flex": -1,
            "textAlign": "left",
            "textAlignVertical": "top",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="" />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "black",
            "borderWidth": 1,
            "height": 20,
            "justifyContent": "center",
            "marginTop": 50,
            "width": 100,
          },
          Object {
            "alignItems": "center",
            "borderColor": "black",
            "borderWidth": 1,
            "height": 35,
            "justifyContent": "center",
            "marginTop": 50,
            "width": 100,
          },
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontSize": 30,
            },
          ]
        }>
        LOGIN
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderColor": "transparent",
        "borderWidth": 1,
        "height": 35,
      }
    } />
</View>
`;

exports[`test renders submitting submitFailed, error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": -1,
    }
  }>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
        },
        Object {
          "marginLeft": 10,
          "marginRight": 10,
        },
        undefined,
        Object {
          "flexDirection": "row",
        },
      ]
    }>
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="username"
      style={
        Array [
          Object {
            "backgroundColor": "white",
          },
          Object {
            "height": 40,
            "width": 400,
          },
          Object {},
          Object {
            "flex": -1,
            "textAlign": "left",
            "textAlignVertical": "top",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="" />
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
        },
        Object {
          "marginLeft": 10,
          "marginRight": 10,
        },
        undefined,
        Object {
          "flexDirection": "row",
        },
      ]
    }>
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="password"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
          },
          Object {
            "height": 40,
            "width": 400,
          },
          Object {},
          Object {
            "flex": -1,
            "textAlign": "left",
            "textAlignVertical": "top",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="" />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "black",
            "borderWidth": 1,
            "height": 20,
            "justifyContent": "center",
            "marginTop": 50,
            "width": 100,
          },
          Object {
            "alignItems": "center",
            "borderColor": "black",
            "borderWidth": 1,
            "height": 35,
            "justifyContent": "center",
            "marginTop": 50,
            "width": 100,
          },
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 15,
            },
            Object {
              "fontSize": 30,
            },
          ]
        }>
        LOGIN
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderColor": "transparent",
        "borderWidth": 1,
        "height": 35,
      }
    } />
</View>
`;
